demos/portal-embed-demo/public/js/portalog/portals-controller.js

```diff
@@ -0,0 +1,239 @@
+        this.root.appendChild(this.portal);
+        this.root.appendChild(messageUI);
+
+        // Add animation on-click
+        this.portal.addEventListener('click', (evt) => {
+            this.initialY = this.root.getBoundingClientRect().y;
+            this.initialWidth = this.root.getBoundingClientRect().width;
+            this._remove(this.root.querySelector('.message-controller'));
+            this.root.style.transition =
+                `top 0.6s cubic-bezier(.49,.86,.37,1.01),
+                 left 0.3s cubic-bezier(.49,.86,.37,1.01), 
+                width 0.3s cubic-bezier(.49,.86,.37,1.01),
+                padding-top 0.3s cubic-bezier(.49,.86,.37,1.01)`;
+            this.root.style.top = `${this.anmiateY - this.initialY}px`;
+            this.root.style.width = '95%';
+            this.root.style.paddingTop = 'calc(95% * 0.65)';
+            this.root.style.left = 'calc((100% - 95%)/2)';
+            this.portal.postMessage({ control: 'hide' }, this.origin);
+        });
+
+    }
+
+    /**
+     * Adding event listeners
+     */
+    hookEvents() {
+        
+        // Event fires when coming back from TTT Archive
+        window.addEventListener('portalactivate', (evt) => {
+            // reset the overflow style to default
+            document.body.style.overflow = '';
+            // embed predecessor
+            this.embed(evt.adoptPredecessor());
+        });
+
+        // Event fires after the portal on-click animation finishes
+        this.root.addEventListener('transitionend', (evt) => {
+            if (evt.propertyName === 'top') {
+                const isFollowed = document.querySelector('#follow')
+                    .classList.contains('followed');
+
+                // Activate portal with data used in the activated page
+                this.portal.activate({
+                    data: {
+                        followed: isFollowed,
+                        name: 'Yusuke Utsunomiya',
+                        photoSrc: '/img/profile.png',
+                        initialY: this.initialY,
+                        activatedWidth: this.root.getBoundingClientRect().width,
+                        initialWidth: this.initialWidth,
+                    }
+                }).then((_) => {
+                    // Resolves with undefined when adoped as a predecessor
+                    // Check if window.portalHost is present (just in case)
+                    if(!window.portalHost){
+                        return;
+                    }
+
+                    // hide the scroll bar
+                    document.body.style.overflow = 'hidden';
+
+                    // Listen to messages (follow/unfollow)
+                    window.portalHost.addEventListener('message', (evt) => {
+                        const isFollowed = evt.data.isFollowed;
+                        this._changeFollowStatus(!isFollowed);
+                    });
+                });
+
+                this._resetRoot();
```

Comment:
It's not obvious what it means to "reset the root". It looks like this resets the animated properties, so rename and update function comment accordingly.

Reply:
Changed it to:
```javascript
// Reset the position of the container after the portal activates
this._resetPositionOfEmbedContainer();
```